client/src/game/ui/reactive/view/statsView.ts

```diff
@@ -52,6 +52,18 @@ export function onEfficiencyChanged(data: {
     cardsGotten += globals.efficiencyModifier;
     cardsGottenModified = true;
   }
+
+  let cardsGottenByNotes: number | string;
+  if (!globals.state.finished && globals.state.playing) {
+    cardsGottenByNotes = 0;
+    for (const card of globals.deck) {
+      cardsGottenByNotes += card.notesEfficiencyModifier();
+    }
+    cardsGotten += cardsGottenByNotes;
+  } else {
+    cardsGottenByNotes = "-";
+  }
```

Comment:
Typically we don't access globals from code that is inside a view. I think this should be refactored to be a field in `StatsState` and be calculated from `statsReducer`. At that point we would also need a new action type for changing a note so that `statsReducer` will get called appropriately. Do you think you could do that? I could also do that as a follow-up.

We don't access globals from views because a view needs to be rerun every time that a value it depends on changes Imagine that `cardsGottenByNotes` is nonzero and the game is terminated. `onEfficiencyChanged` will not be called, but it needs to be called so that it recalculates the values with `globals.state.finished === true` such that it ignores notes.

Reply:
Makes sense, I'll give it a shot this weekend.

Reply:
Awesome. Nice PR btw. Thanks for your work